### PR TITLE
HTTP storage: Fix IPv6 address handling

### PR DESCRIPTION
**Pull request by gjasny**

### CODE REVIEW

File: LICENSE.adoc [hunk 1]
```diff
@@ -678,7 +678,7 @@ src/third_party/httplib.h
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 cpp-httplib - A C++11 single-file header-only cross platform HTTP/HTTPS library.
-Copied from commit 6b9ffc8b of https://github.com/yhirose/cpp-httplib[cpp-httplib].
+Copied from commit 06bfa7e0 of https://github.com/yhirose/cpp-httplib[cpp-httplib].
```
Comment (jrosdahl) [inline LICENSE.adoc:681]: Looks like `httplib.*` are identical to those in the later release v0.9.1, so I suggest:

```suggestion
Copied from https://github.com/yhirose/cpp-httplib[cpp-httplib] v0.9.1.
```

File: src/storage/secondary/HttpStorage.cpp [hunk 2]
```diff
@@ -132,7 +155,10 @@ HttpStorage::HttpStorage(const Url& url, const AttributeMap& attributes)
     m_http_client(make_client(url))
 {
   m_http_client->set_default_headers(
-    {{"User-Agent", FMT("{}/{}", CCACHE_NAME, CCACHE_VERSION)}});
+    {// explicit setting of the Host header is required due to IPv6 address
+     // handling issues in httplib
```
Comment (jrosdahl) [inline src/storage/secondary/HttpStorage.cpp:159]: Is this something we should report to cpp-httplib?

File: src/storage/secondary/HttpStorage.cpp [hunk 3]
```diff
@@ -85,18 +85,41 @@ get_url_path(const Url& url)
...
+  Url host_and_port_only;
+  host_and_port_only.host(url.host(), url.ip_version()).port(url.port());
+
+  // The rendered_value now contains a string like '//[::1]:8080'. The trailing
+  // slashes must be stripped.
+  const auto rendered_value = host_and_port_only.str();
+  const auto prefix = nonstd::string_view{"//"};
+  if (!Util::starts_with(rendered_value, prefix)) {
+    throw Error("Expected partial URL to start with '//': '{}'",
+                rendered_value);
```
Comment (jrosdahl) [inline src/storage/secondary/HttpStorage.cpp:104]:
```suggestion
    throw Error(
      "Expected partial URL to start with '{}': '{}'", prefix, rendered_value);
```

File: src/storage/secondary/HttpStorage.cpp (hunk 2 above)
Comment (gjasny) [reply src/storage/secondary/HttpStorage.cpp:159]: I mentioned it on the closed PR but opened a new issue: yhirose/cpp-httplib#998

**Comment by jrosdahl** [comment]
Thanks.

**Pull request merged**